src/Stats.PostProcessReports/DetailedReportPostProcessor.cs

```diff
@@ -0,0 +1,371 @@
+            {
+                var blobName = GetBlobName(sourceBlob);
+                var workBlobUri = _workStorage.ResolveUri(blobName);
+                var sourceBlobStats = new BlobStatistics();
+                var individualReports = await ProcessSourceBlobAsync(sourceBlob, sourceBlobStats, totals);
+                using (_logger.BeginScope("Processing {BlobName}", blobName))
+                {
+                    if (individualReports != null && individualReports.Any())
+                    {
+                        var consumerTasks = Enumerable
+                            .Range(1, _configuration.ReportWriteDegreeOfParallelism)
+                            .Select(instanceId => WriteReports(instanceId, individualReports, sourceBlobStats, blobName, cancellationToken))
+                            .ToList();
+
+                        await Task.WhenAll(consumerTasks);
+                        ++totals.SourceFilesProcessed;
+                        totals.TotalLinesProcessed += sourceBlobStats.TotalLineCount;
+                        totals.TotalLinesFailed += sourceBlobStats.LinesFailed;
+                        totals.TotalFilesCreated += sourceBlobStats.FilesCreated;
+                        var metadata = new Dictionary<string, string>
+                        {
+                            { TotalLinesMetadataItem, sourceBlobStats.TotalLineCount.ToString() },
+                            { LinesFailedMetadataItem, sourceBlobStats.LinesFailed.ToString() },
+                            { FilesCreatedMetadataItem, sourceBlobStats.FilesCreated.ToString() },
+                        };
+                        await _workStorage.SetMetadataAsync(workBlobUri, metadata);
+                        _logger.LogInformation(
+                            "Finished processing {BlobName}: total lines: {TotalLines}, created {FilesCreated} files, failed to parse {FailedLines} lines",
+                            blobName,
+                            sourceBlobStats.TotalLineCount,
+                            sourceBlobStats.FilesCreated,
+                            sourceBlobStats.LinesFailed);
+                        _telemetryService.ReportFileProcessed(sourceBlobStats.TotalLineCount, sourceBlobStats.FilesCreated, sourceBlobStats.LinesFailed);
+                    }
+                }
+            }
+            var jobSucceededUrl = _workStorage.ResolveUri(JobSucceededFilename);
+            var jobSucceededContent = new StringStorageContent("", TextContentType);
+            await _workStorage.Save(jobSucceededUrl, jobSucceededContent, overwrite: true, cancellationToken: cancellationToken);
+            _telemetryService.ReportTotals(totals.SourceFilesProcessed, totals.TotalLinesProcessed, totals.TotalFilesCreated, totals.TotalLinesFailed);
+        }
+
+        private async Task<bool> SourceBlobExistsInWorkLocationAsync(List<StorageListItem> jsonBlobs, CancellationToken cancellationToken)
```

Comment:
```suggestion
        private async Task<bool> AnySourceBlobExistsInWorkLocationAsync(List<StorageListItem> jsonBlobs, CancellationToken cancellationToken)
```
? might be clearer this is just a single blob check